### PR TITLE
Web3Signer: Object Mapper and Types

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Feature

Part of #9994

**What does this PR do? Why is it needed?**

This adds the JSON object types for web3signer and the object mapper functions to map from Prysm Datatypes to Web3signer datatypes

### CODE REVIEW

File: validator/keymanager/remote-web3signer/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - SYNC_COMMITTEE_CONTRIBUTION_AND_PROOF <- *validatorpb.SignRequest_ContributionAndProof
+- Reload Keys: reloads all public keys from the web3signer.
+- Get Server Status: returns OK if the web3signer is ok.
+
+## Files Added and Files Changed
+
+- Files Added:
+    - validator/keymanager/remote-web3signer package
+
+- Files Modified:
```
Comment (james-prysm) [inline validator/keymanager/remote-web3signer/README.md:45]: this is actually part of the cli pr so please ignore this section, it was just to keep track of file changes as we might want to refactor this in the future

File: validator/keymanager/remote-web3signer/client_test.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	"net/http"
 	"testing"
 
+	v1 "github.com/prysmaticlabs/prysm/validator/keymanager/remote-web3signer/v1"
+
```
Comment (rauljordan) [inline validator/keymanager/remote-web3signer/client_test.go:12]:
```suggestion
```

File: validator/keymanager/remote-web3signer/keymanager.go [hunk 3]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"fmt"
 
+	v1 "github.com/prysmaticlabs/prysm/validator/keymanager/remote-web3signer/v1"
+
```
Comment (rauljordan) [inline validator/keymanager/remote-web3signer/keymanager.go:8]:
```suggestion
```

File: validator/keymanager/remote-web3signer/v1/custom_mappers.go [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+			block.ParentRoot,
+		),
+		StateRoot: hexutil.Encode(
+			block.StateRoot,
+		),
+	}, nil
+}
+
+// MapBeaconBlockBodyAltair maps the eth2.BeaconBlockBodyAltair proto to the Web3Signer spec.
+func MapBeaconBlockBodyAltair(body *ethpb.BeaconBlockBodyAltair) (*BeaconBlockBodyAltair, error) {
```
Comment (rauljordan) [inline validator/keymanager/remote-web3signer/v1/custom_mappers.go:299]: Disregard my comment above, saw that you included Altair and Merge

File: validator/keymanager/remote-web3signer/v1/custom_mappers.go [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+			BlockHash:    hexutil.Encode(body.Eth1Data.BlockHash),
+		},
+		Graffiti:          hexutil.Encode(body.Graffiti),
+		ProposerSlashings: make([]*ProposerSlashing, len(body.ProposerSlashings)),
+		AttesterSlashings: make([]*AttesterSlashing, len(body.AttesterSlashings)),
+		Attestations:      make([]*Attestation, len(body.Attestations)),
+		Deposits:          make([]*Deposit, len(body.Deposits)),
+		VoluntaryExits:    make([]*SignedVoluntaryExit, len(body.VoluntaryExits)),
+		SyncAggregate: &SyncAggregate{
+			SyncCommitteeBits:      hexutil.Encode(body.SyncAggregate.SyncCommitteeBits),
```
Comment (rauljordan) [inline validator/keymanager/remote-web3signer/v1/custom_mappers.go:318]: I think SyncCommitteeBits being a special type, (Bitvector), not sure if the hexutil will directly represent it as we need. You might want to call the .Bytes() method of bitvectors and bitlists https://github.com/prysmaticlabs/go-bitfield/blob/385d8c5e3fb7f24c83e25f32dd9511f5462ea0b8/bitvector512.go#L73 before doing hexutil.Encode

File: validator/keymanager/remote-web3signer/v1/web3signer_types.go [hunk 6]
```diff
@@ -0,0 +1,326 @@
+package v1
+
+/* Web3Signer Specs are found on the following link: https://consensys.github.io/web3signer/web3signer-eth2.html */
```
Comment (rauljordan) [inline validator/keymanager/remote-web3signer/v1/web3signer_types.go:3]: Links can go stale, or become broke in the future. Recommend to just ask the user to look up Consensys' Web3Signer API specification and that these types represent version 1.0

File: validator/keymanager/remote-web3signer/client.go [hunk 7]
```diff
@@ -10,6 +10,8 @@ import (
 	"net/url"
 	"time"
 
+	v1 "github.com/prysmaticlabs/prysm/validator/keymanager/remote-web3signer/v1"
+
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply validator/keymanager/remote-web3signer/client.go:14]: cool added those configs

Comment (rauljordan) [inline validator/keymanager/remote-web3signer/client.go:14]:
```suggestion
```

**Review by rauljordan — DISMISSED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**